utils/filesystem/interfaces.go

```diff
@@ -200,10 +200,18 @@ type FS interface {
 	WalkWithContextAndExclusionPatterns(ctx context.Context, root string, fn filepath.WalkFunc, exclusionPatterns ...string) error
 	// Ls lists all files and directory (equivalent to ls)
 	Ls(dir string) (files []string, err error)
+	// LsRecursive lists all files and directory recursively, including subdirectories
+	LsRecursive(ctx context.Context, dir string) (files []string, err error)
 	// LsWithExclusionPatterns lists all files and directory (equivalent to ls) but exclude the ones matching the exclusion patterns.
 	LsWithExclusionPatterns(dir string, exclusionPatterns ...string) (files []string, err error)
+	// LsRecursiveWithExtensionPatterns lists all files and directory recursively, including subdirectories but exclude the ones matching the exclusion patterns.
+	LsRecursiveWithExtensionPatterns(ctx context.Context, dir string, exclusionPatterns ...string) (files []string, err error)
+	// LsRecursiveWithExtensionPatternsAndLimits lists all files and directory recursively, including subdirectories but exclude the ones matching the exclusion patterns and add some limits for recursion
+	LsRecursiveWithExtensionPatternsAndLimits(ctx context.Context, dir string, limit ILimits, exclusionPatterns ...string) (files []string, err error)
 	// LsFromOpenedDirectory lists all files and directory (equivalent to ls)
 	LsFromOpenedDirectory(dir File) (files []string, err error)
+	// LsRecursiveFromOpenedDirectory lists all files and directory recursively
```

Comment:
your code is not including directories in it's output so this description is wrong is wrong:

```
	if !info.IsDir() {
			files = append(files, path)
		}
```